capellambse/model/crosslayer/capellacore.py

```diff
@@ -21,11 +24,153 @@ class Constraint(c.GenericElement):
 
 @c.xtype_handler(None)
 class Generalization(c.GenericElement):
-    """A Generalization"""
+    """A Generalization."""
 
     _xmltag = "ownedGeneralizations"
 
 
+@c.xtype_handler(None)
+class EnumerationPropertyLiteral(c.GenericElement):
+    """A Literal for EnumerationPropertyType."""
+
+    _xmltag = "ownedLiterals"
+
+
+@c.xtype_handler(None)
+class EnumerationPropertyType(c.GenericElement):
+    """An EnumerationPropertyType."""
+
+    _xmltag = "ownedEnumerationPropertyTypes"
+
+    literals = c.DirectProxyAccessor(
+        EnumerationPropertyLiteral, aslist=c.ElementList
+    )
+
+
+# Following classes belong to the PVMT Extension
+
+
+class PropertyValue(c.GenericElement):
+    """Abstract base class for PropertyValues."""
+
+    _xmltag = "ownedPropertyValues"
+
+    enumerations = c.DirectProxyAccessor(
+        EnumerationPropertyType, aslist=c.ElementList
+    )
+
+    value: xmltools.AttributeProperty | c.AttrProxyAccessor
+
+
+@c.xtype_handler(None)
+class StringPropertyValue(PropertyValue):
+    """A string property value."""
+
+    value = xmltools.AttributeProperty("_element", "value", default="")
+
+
+@c.xtype_handler(None)
+class BooleanPropertyValue(PropertyValue):
+    """A boolean property value."""
+
+    value = xmltools.BooleanAttributeProperty("_element", "value")
+
+
+@c.xtype_handler(None)
+class IntegerPropertyValue(PropertyValue):
+    """An integer property value."""
+
+    value = xmltools.AttributeProperty(
+        "_element", "value", returntype=int, default=0
+    )
+
+
+@c.xtype_handler(None)
+class FloatPropertyValue(PropertyValue):
+    """A floating point property value."""
+
+    value = xmltools.AttributeProperty(
+        "_element", "value", returntype=float, default=0.0
+    )
+
+
+@c.xtype_handler(None)
+class EnumerationPropertyValue(PropertyValue):
+    """An enumeration property value."""
+
+    type = c.AttrProxyAccessor(EnumerationPropertyType, "type")
+    value = c.AttrProxyAccessor(EnumerationPropertyLiteral, "value")
+
+
+@c.xtype_handler(None)
+class PropertyValueGroup(c.GenericElement):
+    """A group for PropertyValues."""
+
+    _xmltag = "ownedPropertyValueGroups"
+
+    properties = c.DirectProxyAccessor(
+        c.GenericElement,
+        (
+            StringPropertyValue,
+            BooleanPropertyValue,
+            IntegerPropertyValue,
+            FloatPropertyValue,
+            EnumerationPropertyValue,
+        ),
+        aslist=c.ElementList,
+    )
+
+    @property
+    def scope(self) -> str | list[str] | None:
+        """Valid architecture layer(s) for this ``PropertyValueGroup``."""
+        if self.description is not None:
+            group_scopes = pvmt.validation.yield_group_scope(self.description)
+            for scope_type, scope in group_scopes:
+                if scope_type == "ARCHITECTURE":
+                    return scope.split(";") if ";" in scope else scope
+
+        return self.description
```

Comment:
The "scope" is a) much more than just the valid architecture layers (it encompasses all validation rules, and as a user I would expect to be able to access and modify all of them) and b) defined by the PVMT extension (as far as I'm aware), and should therefore be part of `capellambse.extensions.pvmt` instead.